### PR TITLE
RavenDB-21450 Add badges and limit restrictions to Sorters & Analyzers

### PR DESCRIPTION
**Pull request by mateuszbartosik**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-21450

### Additional description
Added badges and limit restrictions to Sorters & Analyzers (both db and server-wide)

### Type of change
- Optimization

### How risky is the change?
- Low

### Backward compatibility
- Non breaking change

### Is it platform specific issue?
- No

### Documentation update
- No documentation update is needed 

### Testing by Contributor
- It has been verified by manual testing

### Testing by RavenDB QA team
- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

**Comment by mateuszbartosik** [comment]
Will need a rebase after https://github.com/ravendb/ravendb/pull/17408 merge

File: src/Raven.Studio/typescript/components/pages/database/settings/customAnalyzers/DatabaseCustomAnalyzers.tsx [hunk 1]
```diff
@@ -77,156 +79,186 @@ export default function DatabaseCustomAnalyzers({ db }: NonShardedViewProps) {
...
+                        ({numberOfCustomAnalyzersInCluster}/{licenseClusterLimit})
+                    </strong>
+                    <br /> Delete unused analyzers or{" "}
+                    <strong>
+                        <a href={upgradeLicenseLink} target="_blank">
+                            upgrade your license
+                        </a>
+                    </strong>
+                </Alert>
+            )}
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/pages/database/settings/customAnalyzers/DatabaseCustomAnalyzers.tsx:124]: Should there also be a database limit alert? Or we leave a badge?

File: src/Raven.Studio/typescript/components/pages/database/settings/customAnalyzers/DatabaseCustomAnalyzers.tsx [hunk 2]
```diff
@@ -49,6 +50,7 @@ export default function DatabaseCustomAnalyzers({ db }: NonShardedViewProps) {
     const licenseClusterLimit = useAppSelector(licenseSelectors.statusValue("MaxNumberOfCustomAnalyzersPerCluster"));
     const licenseDatabaseLimit = useAppSelector(licenseSelectors.statusValue("MaxNumberOfCustomAnalyzersPerDatabase"));
     const numberOfCustomAnalyzersInCluster = useAppSelector(licenseSelectors.limitsUsage).NumberOfAnalyzersInCluster;
+    const hasServerWideCustomAnalyzers = useAppSelector(licenseSelectors.statusValue("HasServerWideAnalyzers"));
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/pages/database/settings/customAnalyzers/DatabaseCustomAnalyzers.tsx:53]: We should override it in useLimitedFeatureAvailability below

**Review by kalczur — CHANGES_REQUESTED** [review]

File: src/Raven.Studio/typescript/components/pages/database/settings/customAnalyzers/DatabaseCustomAnalyzers.tsx (hunk 1 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/pages/database/settings/customAnalyzers/DatabaseCustomAnalyzers.tsx:124]: I've added the database limit alert to make it consistent with other views

File: src/Raven.Studio/typescript/components/pages/database/settings/customAnalyzers/DatabaseCustomAnalyzers.tsx (hunk 2 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/pages/database/settings/customAnalyzers/DatabaseCustomAnalyzers.tsx:53]: Override added

**Review by kalczur — APPROVED** [review]

**Review by ml054 — APPROVED** [review]

**Pull request merged**